File: backend/authn/mcp_scopes.py
"""MCP tool scope constants - reuses the exact same scope strings REST already enforces
(OAUTH2_PROVIDER["SCOPES"] in config/settings.py), not a parallel taxonomy. Mirrors the Java
backend's McpScopes.java. ``ACTIVITIES_READ`` is the umbrella "read my training data" scope for
every read tool across every domain (activities/workouts/calendar/gear/races/zones/etc.) - no
per-domain read scopes exist, same product choice as Java's.
"""

ACTIVITIES_READ = "activities:read"
ACTIVITIES_WRITE = "activities:write"
WORKOUTS_WRITE = "workouts:write"
CALENDAR_WRITE = "calendar:write"
GEAR_WRITE = "gear:write"

File: backend_java/src/main/java/com/cadence/api/mcp/dispatch/McpScopes.java
package com.cadence.api.mcp.dispatch;

/**
 * The same six scope strings {@code FirstPartyClientConfig}/{@code McpClientConfig} already
 * use - not a new taxonomy. {@code activities:read} is reused as the umbrella "read my training
 * data" scope for every MCP read tool (workouts/calendar/gear/races included), since the
 * existing scope set has no per-domain read scopes - see the plan's "read-scope granularity"
 * open question.
 */
public final class McpScopes {

	public static final String ACTIVITIES_READ = "activities:read";
	public static final String ACTIVITIES_WRITE = "activities:write";
	public static final String WORKOUTS_WRITE = "workouts:write";
	public static final String CALENDAR_WRITE = "calendar:write";
	public static final String GEAR_WRITE = "gear:write";

	private McpScopes() {
	}
}
